### DISCUSSION TITLE
Get Depth Image Without RGB in PyBullet

### DISCUSSION THREAD

**Discussion by gmargo11** [bulletphysics/bullet3#3890] [Q&A]
Is there any way to turn off RGB image construction on getCameraImage()? I would like to speed up my code by only querying depth images.

**Comment by erwincoumans** [comment]
This isn't implemented at the moment, but it would be useful. It is on my todo list.

In a nutshell, we can disable the fragment shader code, and add some plumbing to enable this (or comment out some code).

Assuming you are using the default CPU/TinyRenderer, the code is here:
See https://github.com/bulletphysics/bullet3/blob/master/examples/TinyRenderer/our_gl.cpp#L203
and
https://github.com/bulletphysics/bullet3/blob/master/examples/TinyRenderer/our_gl.cpp#L148

For hardware OpenGL, the benefits are likely smaller. We should use a simpler shader, in Bullet/examples/OpenGLWindow folder https://github.com/bulletphysics/bullet3/tree/master/examples/OpenGLWindow
